src/components/OSMElementData/OSMElementData.jsx

```diff
@@ -0,0 +1,196 @@
+import _get from 'lodash/get'
+import _flatten from 'lodash/flatten'
+import AsMappableTask
+       from '../../interactions/Task/AsMappableTask'
+import AsIdentifiableFeature
+       from '../../interactions/TaskFeature/AsIdentifiableFeature'
+import Dropdown from '../Dropdown/Dropdown'
+import SvgSymbol from '../SvgSymbol/SvgSymbol'
+import BusySpinner from '../BusySpinner/BusySpinner'
+import messages from './Messages'
+
+const OSM_SERVER = window.env.REACT_APP_OSM_SERVER
+
+const OSMElementData = props => {
+  const [selectedFeatureId, setSelectedFeatureId] = useState(null)
+  const [element, setElement] = useState(null)
+  const [fetchingElement, setFetchingElement] = useState(null)
+  const [failedElement, setFailedElement] = useState(null)
+  const [fetchErr, setFetchErr] = useState(null)
+
+  const { fetchOSMElement, taskBundle } = props
+  const primaryTask = props.task
+  const allTasks = taskBundle ? taskBundle.tasks : [primaryTask]
+
+  const featureIds = _flatten(_compact(_map(allTasks, task => {
+    const geometries = AsMappableTask(task).normalizedGeometries()
+    return geometries ? _compact(_map(
+      geometries.features,
+      f => AsIdentifiableFeature(f).normalizedTypeAndId(true, '/')
+    )) : null
+  })))
+
+  // If the selected feature is no longer available (e.g. because a task bundle
+  // was unbundled), then clear the selection
+  useEffect(() => {
+    if (selectedFeatureId && featureIds.indexOf(selectedFeatureId) === -1) {
+      setSelectedFeatureId(null)
+    }
+  }, [featureIds, selectedFeatureId])
+
+  // Fetch and process the OSM Data
+  useEffect(() => {
+    const activeFeatureId = selectedFeatureId ? selectedFeatureId : featureIds[0]
+
+    // If we're already fetching data for the active feature, or if the fetch failed,
+    // there's nothing to do
+    if (fetchingElement === activeFeatureId || failedElement === activeFeatureId) {
+      return
+    }
+
+    else if (_isEmpty(featureIds)) {
+      // TODO: should I call setElement(null)?
+      // No features to fetch, so nothing to do
+      return
+    } else if (element) {
+      return
+    }
+
+    setFetchingElement(activeFeatureId)
+    setFailedElement(null)
+    setFetchErr(null)
+    fetchOSMElement(activeFeatureId).then(element => {
+      // If for some reason we don't get any element entries, record a failure
+      // to prevent further attempts to refetch
+      if (!element) { // TODO Can this happen ?
+        setFailedElement(activeFeatureId)
+        setFetchingElement(null)
+        return
+      }
+
+      // Sort history entries by version, reversing to get descending order
+      setElement(element) // TODO: Should I store it to avoid other calls when the user switch the feature?
+      setFetchingElement(null)
+    }).catch(err => {
+      setFailedElement(activeFeatureId)
+      setFetchErr(err)
+      setFetchingElement(null)
+    })
+  }, [selectedFeatureId, featureIds, fetchingElement, failedElement, fetchOSMElement])
```

Comment:
I am not entirerly sure about those dependencies, but those are the ones in `OSMElementHistory` component.

Reply:
I went with `useQuery` so I do not need all those dependencies.